platform/binance/block.go

```diff
@@ -2,38 +2,93 @@ package binance
 
 import (
 	"github.com/trustwallet/blockatlas/pkg/blockatlas"
-	"github.com/trustwallet/blockatlas/pkg/errors"
+	"github.com/trustwallet/blockatlas/pkg/numbers"
+	"time"
 )
 
 func (p *Platform) CurrentBlockNumber() (int64, error) {
-	// No native function to get height in explorer API
-	// Workaround: Request list of blocks
-	// and return number of the newest one
-	list, err := p.client.GetBlockList(1)
+	info, err := p.rpcClient.fetchNodeInfo()
 	if err != nil {
 		return 0, err
 	}
-	if len(list.BlockArray) == 0 {
-		return 0, errors.E("no block descriptor found", errors.TypePlatformApi)
-	}
-	return list.BlockArray[0].BlockHeight, nil
+
+	return info.SyncInfo.LatestBlockHeight, nil
 }
 
 func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
-	srcTxs, err := p.client.GetBlockByNumber(num)
+	blockTxs, err := p.rpcClient.GetBlockTransactions(num)
 	if err != nil {
 		return nil, err
 	}
+	childTxs := make([]DexTx, 0)
+	for _, bTrx := range blockTxs {
+		childTxs = append(childTxs, normalizeBlockSubTx(&bTrx))
+	}
+
+	var normTxs []blockatlas.Tx
+	print(len(childTxs))
+	for _, srcTx := range childTxs {
+		normT, ok := NormalizeTx(srcTx, "", "")
+		if !ok {
+			continue
+		}
+		normTxs = append(normTxs, normT...)
+	}
+
+	return &blockatlas.Block{Number: num, Txs: normTxs}, nil
+}
+
+// Normalize block sub transaction from RPC to explorer transaction
+func normalizeBlockSubTx(t *TxV2) DexTx {
```

Comment:
Also you don't change TxV2 in this func
You create a `tx` object and modify
Imho `normalizeBlockSubTx(txV2 TxV2) DexTx {` or `normalizeBlockSubTx(txV2 TxV2) *DexTx {` 
will be better
What do you think?